connectors/qualys/connector/inc_import.py

```diff
@@ -0,0 +1,168 @@
+        from CAR DB and data source asset information
+        params: asset(str): host asset response
+        """
+
+        context().logger.debug('Update vertices started')
+
+        asset_id = asset['HostAsset']['id']
+
+        # Ignore the newly created assets
+        if float(self.last_model_state_id) > get_epoch_time(asset['HostAsset']['created']):
+
+            asset_edges = self.get_active_asset_edges(asset_id)
+
+            if asset['HostAsset'].get('dnsHostName'):
+                asset_edges['hostname'].discard(asset['HostAsset'].get('dnsHostName').lower())
+            elif asset['HostAsset'].get('name'):
+                asset_edges['hostname'].discard(asset['HostAsset']['name'].lower())
+
+            # remove active account from edge list
+            if asset['HostAsset'].get('account'):
+                for account in asset['HostAsset']['account']['list']:
+                    asset_edges['account'].discard(account['HostAssetAccount']['username'])
+
+            # remove active geo location from edge list
+            for row in asset['HostAsset']['sourceInfo']['list']:
+                for asset_location in row:
+                    asset_edges['geolocation'].discard(find_location(row[asset_location]))
+
+            # remove active ipaddress, macaddress from edge list
+            network_interface = asset['HostAsset'].get('networkInterface', {})
+            for interface in network_interface.get('list', {}):
+                asset_edges['ipaddress'].discard(interface['HostAssetInterface']['address'])
+                key = interface['HostAssetInterface'].get('macAddress', "")
+                if not key:
+                    continue
+                asset_edges['macaddress'].discard(key)
+
+            # remove active vulnerability, application from edge list
+            if 'vmdrVulnList' in asset['HostAsset']:
+                for vuln in asset['HostAsset']['vmdrVulnList']:
+                    for app in asset['HostAsset']['applications']:
+                        if app['productName'].lower() in vuln['RESULTS'].lower():
+                            asset_edges['application'].discard(str(app['id']))
+                    asset_edges['vulnerability'].discard(vuln['QID'])
+
+            for edge_name, edge_ids in asset_edges.items():
+                for edge_id in edge_ids:
+                    self.update_edge.append({'from': asset_id, 'to': edge_id, 'edge_type': 'asset_'+edge_name})
+
+    def get_active_asset_edges(self, asset_id):
+        """
+        Get active edges for an asset
+        params: asset_id(str): asset id
+        return: asset_edges(dict): active asset edges
+        """
+
+        asset_edges = {'ipaddress': set(), 'macaddress': set(), 'hostname': set(),
+                       'account': set(), 'application': set(), 'vulnerability': set(), 'geolocation': set()}
+
+        search_result = context().car_service.graph_search('asset', str(asset_id))

```

Comment:
graph_search is deprecated function shouldn't be used.

Reply:
We didn't find any alternative API for graph search, please let us know which is the alternative API to use for getting asset associated node and edge information if 'graph_search' is deprecated.

Reply:
We will use API 'graph_attribute_search' for getting information on nodes and use the information to disable edges

Reply:
We have updated code to use the API v3 (URL : 'https://***.ibmcloudsecurity.com/api/car/v3/query' ) to get the asset edge information from CAR DB.

Note: not able to use 'graph_attribute_search' to get the edges  since we are getting 500 error code.